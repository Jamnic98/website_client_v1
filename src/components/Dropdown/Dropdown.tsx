import { type ReactNode } from 'react'
import classNames from 'classnames'

import styles from './dropdown.module.css'

export interface DropdownProps {
  items: object[]
  isOpen: boolean
  render: (item: any) => ReactNode
}

export const Dropdown: React.FC<DropdownProps> = ({ isOpen, render, items }) => (
  <div
    className={classNames(styles.dropdown, isOpen ? styles.open : styles.closed)}
    data-testid="application-dropdown"
  >
    {items.map((item, index) => (
      <div key={index} className={styles.dropdownItem}>
        {render(item)}
      </div>
    ))}
  </div>
)
