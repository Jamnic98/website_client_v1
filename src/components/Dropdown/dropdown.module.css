.dropdown {
  z-index: 1;
  width: 120%;
  position: absolute;
  top: 0;
  max-height: 0;
  transition: max-height 0.3s ease;
  overflow: hidden;
  background-color: inherit;
}

.dropdown.open {
  max-height: 50vh;
}

.dropdown.closed {
  z-index: 0;
}

.dropdownItem {
  position: relative;
  height: 100%;
  height: 2.4em;
  text-align: center;
  background-color: inherit;
}
