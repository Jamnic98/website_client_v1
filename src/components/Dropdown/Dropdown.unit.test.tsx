import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

import { Dropdown } from './Dropdown'

type testObject = { letter: string }
const testArray: testObject[] = [
  { letter: 'a' },
  { letter: 'b' },
  { letter: 'c' },
]

describe('Dropdown', () => {
  // draw Dropdown component
  render(
    <Dropdown
      isOpen={false}
      items={testArray}
      render={(item: testObject) => (
        <div data-testid="dropdown-item">{item.letter}</div>
      )}
    />
  )

  const dropdown = screen.getByTestId('application-dropdown')

  it('renders Dropdown', () => {
    expect(dropdown).toBeInTheDocument()
    const dropdownItem = screen.getAllByTestId('dropdown-item')[0]
    expect(dropdown).toContainElement(dropdownItem)
    expect(dropdownItem).toHaveTextContent(testArray[0].letter)
  })

  it('renders Dropdown unchanged', () => {
    expect(
      renderer
        .create(
          <Dropdown
            isOpen={false}
            items={testArray}
            render={() => <div></div>}
          />
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
