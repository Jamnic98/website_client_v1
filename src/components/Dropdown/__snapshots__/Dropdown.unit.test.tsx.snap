// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders Dropdown unchanged 1`] = `
<div
  className="dropdown closed"
  data-testid="application-dropdown"
>
  <div
    className="dropdownItem"
  >
    <div />
  </div>
  <div
    className="dropdownItem"
  >
    <div />
  </div>
  <div
    className="dropdownItem"
  >
    <div />
  </div>
</div>
`;
